pyvista/core/dataset.py

```diff
@@ -2479,9 +2480,122 @@ def find_cells_within_bounds(self, bounds: Iterable[float]) -> np.ndarray:
         locator.FindCellsWithinBounds(list(bounds), id_list)
         return vtk_id_list_to_array(id_list)
 
+    def get_cell(self, i: int) -> Cell:
+        """Return a pyvista.Cell object.
+
+        Parameters
+        ----------
+        i : int
+            Cell ID.
+
+        Returns
+        -------
+        pyvista.Cell
+            The i-th pyvista.Cell.
+
+        Examples
+        --------
+        Get the 0-th cell
+
+        >>> from pyvista import examples
+        >>> mesh = examples.load_airplane()
+        >>> mesh.get_cell(0) # doctest:+SKIP
+        Cell (0x7f6304e0a730)
+          Type:	CellType.TRIANGLE
+          Linear:	True
+          Dimension:	2
+          N Points:	3
+          N Faces:	0
+          N Edges:	3
+          X Bounds:	8.970e+02, 9.075e+02
+          Y Bounds:	4.876e+01, 5.549e+01
+          Z Bounds:	8.075e+01, 8.366e+01
+        """
+        vtkCell = _vtk.vtkGenericCell()
+        self.GetCell(i, vtkCell)
+        return Cell(vtkCell)
+
+    @property
+    def cell(self) -> List[Cell]:
+        """Return a list of cells.
+
+        Returns
+        -------
+        list[pyvista.Cell]
+            A list of `pyvista.Cell` objects.
+
+        Warnings
+        --------
+        The ``edges`` and ``faces`` properties of the :class:`pyvista.Cell`
+        object are generators that should NEVER be converted to a list
+        since repeated calls to the vtk underlying methods can change the
+        returned data.
+        See https://vtk.org/doc/nightly/html/classvtkCell.html#a11b6ba66e9f7e193b204d478379e32ea.
```

Comment:
This warning is not entirely clear to me, and at the same time it's kind of terrifying. Consuming a generator into a list is a very natural thing to do. If this is forbidden, do we really want to keep this feature as-is?

What exactly is the risk? What "underlying methods" do get called and what "returned data" change, exactly? I looked at the link and it wasn't clear to me at all, so I don't think an end user will be better off reading this.

Looking at e.g. the `Cell.faces` property: we seem to just be providing a thin wrapper for `get_face()`. So if the way we do this is fragile, can we just not provide the wrapper, or provide it in a way that is safe?

And this warning should be present in a lot more places (everywhere where a `Cell` might be created, I think). This might overlap with what @MatthewFlamm's already said.

Reply:
It is a vtk limitation.  vtk returns the same object everytime you call get_face() but with new data inside.  So if you make a list, you get a list of length `n_cells` filled with just the last element.  One workaround is to always copy the cell, but this has memory issues.

Reply:
Although, I could be wrong on this, faces and edges are not typically part of a dataset, and it's cells, and would be generated on-the-fly.  So the memory hit would not be so large and would only become a problem if the user created a nested list of all faces for _all_ cells.

Reply:
Thanks, this was my best guess for the situation. Just to make sure we mean the same thing: something like
```py
old_face = cell.get_face(0)
new_face = cell.get_face(1)
new_face is old_face  # oh no
```

But then this means that our generator always returns the same object. And if we did `set(cell.faces)`, we'd end up with a 1-length cell (assuming the object is hashable).

Then my recommendation is not to provide this helper at all. Having a generator that yields the same object each time, given this caveat, is very misleading. Only provide `get_face()` with a huge red warning not to work on two faces at the same time. Or copy, but as you said this is not viable.

Reply:
> Although, I could be wrong on this, faces and edges are not typically part of a dataset, and it's cells, and would be generated on-the-fly. 

I was wondering about this. Cells don't have that many faces [citation needed]. So we could return a copy of each. If a user chooses to hang on to a bunch of cell faces that's not our responsibility (assuming there's no _surprising_ amount of memory overhead, from, I don't know, copying cell faces leading to copying the whole dataset or something).

Reply:
> something like
>```
>old_face = cell.get_face(0)
>new_face = cell.get_face(1)
>new_face is old_face  # oh no
>```

Correct.

> But then this means that our generator always returns the same object. 

Not quite,

```
for face in cell.faces:
    operate_on_face(face)
```

would operate on unique faces.

But I'm tending to agree with you that if we provide the generator option, it will be more natural for users to assume they can make a list.  I'm okay either with explicit clear warnings or removing it.

Reply:
Agreeing with the concerns you raised both @adeak and @MatthewFlamm.
I am also fine with copying, but I am afraid it goes beyond my vtk knowledge. What do you mean exactly ? Using the `DeepCopy()` method of the `vtkCell` ?
Might be unrelated, but I encountered a weird behaviour when using `DeepCopy()`, see https://github.com/pyvista/pyvista/pull/3715#issuecomment-1362178873.

Reply:
Moreover, the pointset test that consistently failed magically passed when merging with last main ? Weird ...

Reply:
> Thanks, this was my best guess for the situation. Just to make sure we mean the same thing: something like
> 
> ```python
> old_face = cell.get_face(0)
> new_face = cell.get_face(1)
> new_face is old_face  # oh no
> ```
> 
> right?

 From this MWE, it seems that the `Cell` faces objects are different:
```python
from pyvista.examples.cells import Tetrahedron

c = Tetrahedron().cell[0]
f1 = c.get_face(0)
f2 = c.get_face(1)
f1 is f2
```
False

Reply:
That looks related, thanks. I'd expect your code to work the way you expect, `DeepCopy` should, you know, do a deep copy. The fact that it doesn't is somewhat concerning, but I'm no VTK expert either.

To see how different the original and the copy are, you can compare `print(cell)` and `print(pvcell)`, there are quite a few attributes that are not carried over.

Reply:
I was also surprised it did not work as expected.

I'm getting slowly convinced that the our best option would be to remove the `faces` and `edges` generators and only provide the `get_face` and `get_edge` methods to end-users, without any terrifying warnings (for which I apologize for :angel:).